[GFC] Move integration coverage to its own file

sammygill opened the pull request:
#### 6a265e4581b6fffff53c4b35782ce2ed6b306f37
<pre>
[GFC] Move integration coverage to its own file
<a href="https://bugs.webkit.org/show_bug.cgi?id=299536">https://bugs.webkit.org/show_bug.cgi?id=299536</a>
<a href="https://rdar.apple.com/161345598">rdar://161345598</a>

Reviewed by Tim Nguyen and Brandon Stewart.

Currently the integration coverage for GFC is contained within a generic
file and lives alongside similar code dedicated for FFC. For better
organization let&apos;s move the grid code to is own dedicated file. We can
then also add AvoidanceReasons and other related bits within that source
file without it worrying about conflicting with others.

Canonical link: <a href="https://commits.webkit.org/300542@main">https://commits.webkit.org/300542@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1688d97b3dc73d6201ce78da3707c2ece0845891

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122938 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42652 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33349 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129585 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75046 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0e21c769-9e7d-4d33-b4d7-574409016869) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124815 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43375 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51246 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93458 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5b592868-3629-4f65-9f81-80361160830b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125889 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34585 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110046 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74075 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/28562d7c-9c00-4fee-946f-da453edcf206) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33565 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28200 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73083 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104293 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28411 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132313 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49887 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37979 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101959 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50264 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106252 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101826 "Found 3 new API test failures: TestWTF:CompletionHandlerDeathTest.MainThreadHandlerOnThreadAssertsDeathTest, WebKitGTK/TestInspector:/webkit/WebKitWebInspector/manual-attach-detach, TestWTF:CompletionHandlerDeathTest.ConstructionThreadHandlerOnThreadAssertsDeathTest (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25874 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47185 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25382 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/46661 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49743 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55503 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49210 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52562 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50892 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WebCore/layout/integration/grid/LayoutIntegrationGridCoverage.cpp:
```diff
@@ -0,0 +1,43 @@
+bool canUseForGridLayout(const RenderGrid& renderGrid)
+{
+    // Grid integration is not enabled yet.
+    if (!renderGrid.document().settings().gridFormattingContextIntegrationEnabled())
```
This if check is currently redundant.

sammygill replied:
Yeah the reason why I added it is so that we could replace the `return false` below soon with some different logic without any risks on trunk

sammygill committed:
[GFC] Move integration coverage to its own file
https://bugs.webkit.org/show_bug.cgi?id=299536
rdar://161345598

Reviewed by Tim Nguyen and Brandon Stewart.

Currently the integration coverage for GFC is contained within a generic
file and lives alongside similar code dedicated for FFC. For better
organization let's move the grid code to is own dedicated file. We can
then also add AvoidanceReasons and other related bits within that source
file without it worrying about conflicting with others.

Canonical link: https://commits.webkit.org/300542@main
